tidy3d/components/geometry.py

```diff
@@ -3724,14 +4090,210 @@ def plot(
+        elif self.operation == "difference":
+            result = ClipOperation.to_polygon_list(shapely.difference(geom_a, geom_b))
+        elif self.operation == "symmetric_difference":
+            result = ClipOperation.to_polygon_list(shapely.symmetric_difference(geom_a, geom_b))
+        else:
+            raise ValueError(
+                "'operation' must be one of 'union', 'intersection', 'difference', or "
+                "'symmetric_difference'."
+            )
+        return result
+
+    @cached_property
+    def bounds(self) -> Bound:  # pylint:disable=too-many-locals
+        """Returns bounding box min and max coordinates.
+
+        Returns
+        -------
+        Tuple[float, float, float], Tuple[float, float float]
+            Min and max bounds packaged as ``(minx, miny, minz), (maxx, maxy, maxz)``.
+        """
+        # Overestimates
+        if self.operation == "difference":
+            result = self.geometry_a.bounds
+        elif self.operation == "intersection":
+            bounds = (self.geometry_a.bounds, self.geometry_b.bounds)
+            result = (
+                tuple(max(b[i] for b, _ in bounds) for i in range(3)),
+                tuple(min(b[i] for _, b in bounds) for i in range(3)),
+            )
+            if any(result[0][i] > result[1][i] for i in range(3)):
+                result = ((0, 0, 0), (0, 0, 0))
+        else:
+            bounds = (self.geometry_a.bounds, self.geometry_b.bounds)
+            result = (
+                tuple(min(b[i] for b, _ in bounds) for i in range(3)),
+                tuple(max(b[i] for _, b in bounds) for i in range(3)),
+            )
+        return result
+
+    def inside(
+        self, x: np.ndarray[float], y: np.ndarray[float], z: np.ndarray[float]
+    ) -> np.ndarray[bool]:
+        """For input arrays ``x``, ``y``, ``z`` of arbitrary but identical shape, return an array
+        with the same shape which is ``True`` for every point in zip(x, y, z) that is inside the
+        volume of the :class:`Geometry`, and ``False`` otherwise.
+
+        Parameters
+        ----------
+        x : np.ndarray[float]
+            Array of point positions in x direction.
+        y : np.ndarray[float]
+            Array of point positions in y direction.
+        z : np.ndarray[float]
+            Array of point positions in z direction.
+
+        Returns
+        -------
+        np.ndarray[bool]
+            ``True`` for every point that is inside the geometry.
+        """
+        inside_a = self.geometry_a.inside(x, y, z)
+        inside_b = self.geometry_b.inside(x, y, z)
+        if self.operation == "union":
+            result = inside_a | inside_b
+        elif self.operation == "intersection":
+            result = inside_a & inside_b
+        elif self.operation == "difference":
+            result = inside_a & ~inside_b
+        elif self.operation == "symmetric_difference":
+            result = inside_a != inside_b
```

Comment:
I think it should be `(inside_a != inside_b) & (inside_a | inside_b)`?

Reply:
I don't see why…
```
| A | B | A!=B | (A!=B) & (A|B) |
| 0 | 0 |  0   |       0        |
| 0 | 1 |  1   |       1        |
| 1 | 0 |  1   |       1        |
| 1 | 1 |  0   |       0        |
```